### PR TITLE
docs: add usage guides for WITOverlayGenerator and WITExtractor

### PR DESCRIPTION
**Pull request by shivanshu877**
Addresses the two open items in https://github.com/swiftwasm/WasmKit/issues/71 (\"Enhance documentation\"):

- Example and usage of `WITOverlayGenerator`
- Example and usage of `WITExtractor`

Both documents are derived from reading the source of the respective modules (`WITOverlayGen.swift`, `WITExtractor.swift`) and their SPM plugins (`WITOverlayPlugin`, `WITExtractorPlugin`).

The intent is the same as the existing module docs in WasmKit's docc bundle — make the WIT tooling discoverable for users coming from outside the project.

## Test plan

- [x] Docs build cleanly with the existing docc setup.
- [ ] A maintainer review on the prose / accuracy is welcome — I worked from the source, but anyone closer to the API will see corner cases I missed.

### CODE REVIEW

File: Documentation/ComponentModel/WITExtractor.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+```
+
+---
+
+## Usage via `wit-tool` CLI
+
+For scripted use, call the `extract-wit` subcommand directly:
+
+```sh
+swift run wit-tool extract-wit \
```
Comment (MaxDesiatov) [inline Documentation/ComponentModel/WITExtractor.md]: It doesn't look like these instructions were verified end-to-end, or at least not verified with the state of the repository as I see it? `swift run wit-tool` won't work. This will fail with `error: no executable product named wit-tool`


```suggestion
swift run WITTool extract-wit \
```

File: Documentation/ComponentModel/WITExtractor.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+---
+
+## End-to-end example
+
+Given a Swift module `MathLib` with the following public API:
+
+```swift
+// Sources/MathLib/MathLib.swift
+public struct Vector2 {
```
Comment (MaxDesiatov) [inline Documentation/ComponentModel/WITExtractor.md:135]: IIRC `WITExtractor` only extracts decls with `@_spi(WIT)` on them.

File: Documentation/ComponentModel/WITExtractor.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+## Usage via `wit-tool` CLI
+
+For scripted use, call the `extract-wit` subcommand directly:
+
+```sh
+swift run wit-tool extract-wit \
+    --swift-api-digester $(xcrun --find swift-api-digester) \
+    --module-name MyLibrary \
+    --package-name my-library \
```
Comment (MaxDesiatov) [inline Documentation/ComponentModel/WITExtractor.md:109]: the plugin form only accepts `--target`, `--sdk`, `--output-mapping`. It auto-derives the WIT package name from `context.package.displayName`. This snippet conflates the plugin and the CLI

**Review by MaxDesiatov — CHANGES_REQUESTED** [review]

**Comment by shivanshu877** [comment]
Thanks @MaxDesiatov — all three addressed in `ef224eb`:

1. **`swift run wit-tool` → `swift run WITTool`** (verified `WITTool` is the actual executable name in `Package.swift`).
2. **End-to-end example no longer conflates plugin and CLI.** Reworked the "Running…" snippet to use the plugin form (`swift package plugin extract-wit --target MathLib`) with a parenthetical pointing at the CLI form for callers that need to override the package name.
3. **Documented the `@_spi(WIT)` opt-in requirement.** Added a callout in the "When to use it" section explaining only `@_spi(WIT)`-annotated declarations are extracted, and applied `@_spi(WIT)` to the example's `Vector2` and `addVectors` so the snippet actually produces the WIT shown below it (verified against `Sources/WITExtractor/SourceSummary.swift` line 221: `guard let spiGroup = decl.body.spi_group_names, spiGroup.contains(\"WIT\") else { return false }`).

Ready for re-review.
